### DISCUSSION TITLE
Location of settings.yml

### DISCUSSION THREAD

**Discussion by rdeago** [statiqdev/Discussions#175] [Q&A] [answered]
I like to keep my web site "sources" (Statiq's RootPath) separate from C# sources (C# project). I have successfully configured RootPath via the bootstrapper factory's ConfigureFileSystem extension method. To my surprise and dismay, however, Statiq does not recognize my `settings.yml` file any longer.

I have isolated the cause in method [Statiq.App.BootstrapperDefaultExtensions.AddDefaultConfigurationFiles](https://github.com/statiqdev/Statiq.Framework/blob/1a8bd58e6f1a4e63f0cae2870db78126bc511928/src/core/Statiq.App/Bootstrapper/BootstrapperDefaultExtensions.cs#L106) where it uses the result of `Directory.GetCurrentDirectory()` as the base path for settings files.

Is this by design? The rationale escapes me. Shouldn't Statiq be looking in RootPath instead?

**Comment by daveaglick** [comment, marked as answer]
The use of the working directory is by design to support the convention of having settings files in the same folder as the project and other .NET application artifacts (this is similar to ASP.NET or other applications that use the `Microsoft.Extensions.Configuration` libraries. This is further solidified by setting the `RunWorkingDirectory` MSBuild prop in the `Statiq.App` targets file here: https://github.com/statiqdev/Statiq.Framework/blob/1a8bd58e6f1a4e63f0cae2870db78126bc511928/src/core/Statiq.App/Statiq.App.targets#L3 - that ensures the working directory is set to the same directory as the project file (so we don't get weird working directory behavior).

So long story short, the convention is to place settings files alongside the .NET project file and the Statiq "root path" is more about where the Statiq `input`, `output`, etc. are found.

Like most things in Statiq, this is easily overridden if you'd like different behavior:

```csharp
await Bootstrapper.Factory
    .CreateDefaultWithout(args, DefaultFeatures.Settings)
    .AddWeb() // Omit if not using Statiq Web
    .ConfigureFileSystem(fs => fs.RootPath = @"[whatever]\new-root")
    .BuildConfiguration(builder => builder
            .SetBasePath(Path.Combine(Directory.GetCurrentDirectory(), "new-root"))
            .AddSettingsFile("settings"))
    .RunAsync();
```

**Comment by rdeago** [comment]
Thanks a lot @daveaglick! `CreateDefaultWithout` is definitely the missing piece; I had already devised a workaround, but it wasn't nearly as clean.

Please bear with me, I've never used ASP.NET (Core or otherwise) so I'm not familiar with its conventions and best practices, and I'm learning as I go.

**Reply by daveaglick** [reply]
No problem at all - the documentation in Statiq around a lot of these things could certainly be much better as well.
